packages/driver/src/cy/snapshots.ts

```diff
@@ -141,64 +147,83 @@ export const create = ($$, state) => {
+    // are no side effects from cloning it. see below for how we re-attach
+    // it to the AUT document
+    // https://github.com/cypress-io/cypress/issues/8679
+    // this can fail if snapshotting before the page has fully loaded,
+    // so we catch this below and return null for the snapshot
+    // https://github.com/cypress-io/cypress/issues/15816
+    const $body = $$(snapshotDocument.importNode($$('body')[0], true))
+    // for the head and body, get an array of all CSS,
+    // whether it's links or style tags
+    // if it's same-origin, it will get the actual styles as a string
+    // it it's cross-domain, it will get a reference to the link's href
+    const { headStyleIds, bodyStyleIds } = snapshotsCss.getStyleIds()
+
+    // replaces iframes with placeholders
+    replaceIframes($body)
+
+    // remove tags we don't want in body
+    $body.find('script,link[rel=\'stylesheet\'],style').remove()
+
+    // here we need to figure out if we're in a remote manual environment
+    // if so we need to stringify the DOM:
+    // 1. grab all inputs / textareas / options and set their value on the element
+    // 2. convert DOM to string: body.prop("outerHTML")
+    // 3. send this string via websocket to our server
+    // 4. server rebroadcasts this to our client and its stored as a property
+
+    // its also possible for us to store the DOM string completely on the server
+    // without ever sending it back to the browser (until its requests).
+    // we could just store it in memory and wipe it out intelligently.
+    // this would also prevent having to store the DOM structure on the client,
+    // which would reduce memory, and some CPU operations
+
+    // now remove it after we clone
+    if (isJqueryElement) {
+      ($elToHighlight as JQuery<HTMLElement>).removeAttr(HIGHLIGHT_ATTR)
+    }
+
+    const $htmlAttrs = getHtmlAttrs($$('html')[0])
 
-      // preserve attributes on the <html> tag
-      const htmlAttrs = getHtmlAttrs($$('html')[0])
-      // the body we clone via importNode above is attached to a transient document
-      // so that there are no side effects from cloning it. we only attach it back
-      // to the AUT document at the last moment (when restoring the snapshot)
-      // https://github.com/cypress-io/cypress/issues/8679
+    return {
+      $body,
+      $htmlAttrs,
+      headStyleIds,
+      bodyStyleIds,
+    }
+  }
+
+  const reifySnapshotBody = (preprocessedSnapshot) => {
+    const $body = preprocessedSnapshot.body.get()
+    const $htmlAttrs = preprocessedSnapshot.htmlAttrs
+
+    return {
+      $body,
+      $htmlAttrs,
+      headStyleIds: null,
+      bodyStyleIds: null,
+    }
+  }
+
+  const createSnapshot = (name, $elToHighlight, preprocessedSnapshot) => {
+    Cypress.action('cy:snapshot', name)
+
+    try {
+      const { $body, $htmlAttrs, headStyleIds, bodyStyleIds } = !preprocessedSnapshot ? createSnapshotBody($elToHighlight) : reifySnapshotBody(preprocessedSnapshot)
```

Comment:
I tried to be as surgical as possible with changing this method. When snapshots are sent back from the spec bridge to the primary, they need to be imported into the primary snapshot document for lookup within the `aut-iframe` for display purposes, as well as the snapshot styles. spec bridge snapshot styles are not mapped by id, but are instead inline in the CSS Map. If no `preprocessedSnapshot` argument is present, the current code path is taken and should NOT change the way snapshots work today outside of multi origin.